narwhals/_duckdb/dataframe.py

```diff
@@ -395,3 +396,52 @@ def drop_nulls(self: Self, subset: list[str] | None) -> Self:
         keep_condition = " and ".join(f'"{col}" is not null' for col in subset_)
         query = f"select * from rel where {keep_condition}"  # noqa: S608
         return self._from_native_frame(duckdb.sql(query))
+
+    def explode(self: Self, columns: str | Sequence[str], *more_columns: str) -> Self:
+        from narwhals.exceptions import InvalidOperationError
+
+        dtypes = import_dtypes_module(self._version)
+
+        to_explode = (
+            [columns, *more_columns]
+            if isinstance(columns, str)
+            else [*columns, *more_columns]
+        )
+        schema = self.collect_schema()
+        for col_to_explode in to_explode:
+            dtype = schema[col_to_explode]
+
+            if dtype != dtypes.List:
+                msg = (
+                    f"`explode` operation not supported for dtype `{dtype}`, "
+                    "expected List type"
+                )
+                raise InvalidOperationError(msg)
+
+        rel = self._native_frame  # noqa: F841
+        columns = self.columns
+        select_unnest_statement = ", ".join(
+            f'unnest("{col}") as "{col}"' if col in to_explode else f'"{col}"'
+            for col in columns
+        )
+        select_null_statement = ", ".join(
+            f'NULL as "{col}"' if col in to_explode else f'"{col}"' for col in columns
+        )
+        where_condition = " and ".join(
+            f'"{col}" is not null and len("{col}") > 0' for col in to_explode
+        )
+
+        query = f"""
+            (
+                select {select_unnest_statement}
+                from rel
+                where {where_condition}
+            )
+            union
+            (
+                select {select_null_statement}
+                from rel
+                where not ({where_condition})
+            );
```

Comment:
Similar logic to the one mentioned in the pyspark PR, since the order should not matter

Reply:
this looks fine, I just wonder if we can use the Python API for this? looks like `duckdb.FunctionExpression('unnest', duckdb.ColumnExpression('a'))` is supported, as well as `rel.union` and `rel.filter`

Reply:
Gave it a try and it would look like the following:


```py
 non_null_rel = rel.filter(
    col_to_explode.isnotnull() & FunctionExpression("len", col_to_explode)>0
    ).select(*[
        FunctionExpression("unnest", col_to_explode).alias(col) if col in columns else col
        for col in original_columns
    ]
)

null_rel = rel.filter(
    ~(col_to_explode.isnotnull() & FunctionExpression("len", col_to_explode)>0)
    ).select(*[
        ConstantExpression(None).alias(col) if col in columns else col
        for col in original_columns
    ]
)

return self._from_native_frame(non_null_rel.union(null_rel))
```